client/src/store/utils/thunkCreators.js

```diff
@@ -117,3 +121,47 @@ export const searchUsers = (searchTerm) => async (dispatch) => {
     console.error(error);
   }
 };
+
+// send new reads to api to update
+const saveMessageReads = async (body) => {
+  const { data } = await axios.put("/api/messages/read", body);
+  return data;
+}
+
+const sendLastReadMessage = (data) => {
+  socket.emit("last-read-message", {
+    message: data.message,
+  });
+};
+
+export const updateLastReadMessage = (body) => async (dispatch) => {
+  // update db
+  const data = await saveMessageReads(body);
+  // update conversations in store from current user
+  // 2nd param "false" signifies that the updateMessageRead is being passed by current user.
+  dispatch(updateMessageRead(data.message, false));
+  
+  //send read update to socket
+  sendLastReadMessage(data);
+}
+
+export const joinChat = (conversation) => async (dispatch) => {
+  await dispatch(setActiveChat(conversation.otherUser.username));
+  setActiveChatInSocket(conversation.otherUser.username);
+  const unreadMessages = conversation.messages.filter((message) => message.senderId === conversation.otherUser.id && message.read === false);
+  const lastUnreadMessage = unreadMessages.length ? unreadMessages.reduce((prev, current) => (prev.id > current.id) ? prev : current) : null;
```

Comment:
I don't quite understand why we're using reduce here. Is the message id a proxy for the date sent? Are the messages not in order at this point?

Reply:
My thinking was best way to get the latest message, regardless of order. I think its safe to assume the messages are in order and could probably use something like unreadMessages[unreadMessages.length-1]